Bluetooth: Mesh: Add LE Pairing Responder model and enable SMP authentication in Distributor sample (#13173)
nrfconnect/sdk-nrf: samples/bluetooth/mesh/dfu/distributor/README.rst

When this sample is used as a Target, it behaves as described in :ref:`ble_mesh_dfu_target_upgrade`.

SMP over Bluetooth authentication
=================================

> By default, when using `SMP over Bluetooth` to access the :ref:`management subsystem <zephyr:mcu_mgr>`, the SMP characteristics don't require authentication.

Reviewer: ```suggestion
By default, the SMP characteristics don't require authentication when using SMP over Bluetooth to access the :ref:`management subsystem <zephyr:mcu_mgr>`.
```

To prevent an unauthenticated access to the device over SMP, it is strongly recommended to enable the :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT_AUTHEN` option.
This will enforce a remote device to initiate a pairing request before accessing SMP characteristics.
See `Zephyr Bluetooth LE Security`_ for more details about securing the Bluetooth LE connection.

> The sample supports :ref:`bt_mesh_le_pair_resp_readme` that allows to pass a passkey over a mesh network when the Distributor has no means to display the passkey.

Reviewer: ```suggestion
The sample supports the :ref:`bt_mesh_le_pair_resp_readme` model that allows sending a passkey over a mesh network when the Distributor has no means of displaying the passkey.
```

> When the model is enabled together with the :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT_AUTHEN` option and a remote device tries to read SMP characteristics, the pairing request will be initiated and the sample will require the remote device to enter the passkey generated by the model.

Reviewer: ```suggestion
When the model and the :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT_AUTHEN` option are enabled while a remote device tries to read the SMP characteristics, the pairing request will be initiated and the sample will require the remote device to enter the passkey generated by the model.
```

> To enable the LE pairing authentication with the LE Pairing Responder model support, set :makevar:`OVERLAY_CONFIG` to :file:`overlay-smp-bt-auth.conf` when building the sample.

Reviewer: ```suggestion
To enable the LE pairing authentication with the LE Pairing Responder model support, set :makevar:`OVERLAY_CONFIG` to :file:`overlay-smp-bt-auth.conf` file when building the sample.
```
